src/server/game/Achievements/AchievementMgr.cpp

```diff
@@ -65,7 +65,7 @@ bool AchievementMgr::CanUpdateCriteriaTree(Criteria const* criteria, CriteriaTre
 {
     AchievementEntry const* achievement = tree->Achievement;
     if (!achievement)
-        return false;
+        return true;
```

Comment:
You are right. Sorry I'm new to TC and still learning how everything works :)
What about the CriteriaHandler check in the new commit?